erpnext/stock/doctype/stock_entry/stock_entry.py

```diff
@@ -126,11 +139,41 @@ def validate_item(self):
 				frappe.throw(_("Row #{0}: Please specify Serial No for Item {1}").format(item.idx, item.item_code),
 					frappe.MandatoryError)
 
+	def validate_qty(self):
+		manufacture_purpose = ["Manufacture", "Material Consumption for Manufacture"]
+
+		if self.purpose in manufacture_purpose and self.work_order:
+			if not frappe.get_value('Work Order', self.work_order, 'skip_transfer'):
+				item_code = []
+				for item in self.items:
```

Comment:
This loop should only be run for raw materials. Add a if condition `if cstr(item.t_warehouse) = '':`